packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -519,6 +520,18 @@ abstract class ImageStreamCompleter with Diagnosticable {
     _checkDisposed();
     _hadAtLeastOneListener = true;
     _listeners.add(listener);
+    _synchronouslyCallAddedListener(listener);
+  }
+
+  /// Add a listener for errors. It is similar to [addListener], but see
```

Comment:
Why is the method called "peek" listener, but the description says it is for listening to errors? Should it be called error listener then? Or the doc should describe what a "peek" is.

Overall, this doc is lacking, see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#writing-prompts-for-good-documentation. Questions that should be answered here: When should I add a peek listener? How is it different from a regular listener (document that here, don't just link to it, it is important information to use this method, if necessary use a macro to avoid repetition)? Can the listener be called synchronously? When does this happen? etc. Also compare this to the existing docs on addListener and see what is useful here: https://master-api.flutter.dev/flutter/painting/ImageStream/addListener.html

Reply:
What happens if I add the same listener multiple times?

Reply:
Ideally, addListener would also link back to this listener (e.g. in a "see also" section) with a short statement about its purpose.

Reply:
Thanks for spotting the outdated doc! It is not a listener for errors, but for (at least) onImage and onError. I will add more doc.